native/acter/src/api/room.rs

```diff
@@ -522,6 +524,85 @@ impl Room {
             .await?
     }
 
+    pub async fn add_parent_room(&self, room_id: String, canonical: bool) -> Result<String> {
+        let room_id = OwnedRoomId::try_from(room_id)?;
+        if !self
+            .get_my_membership()
+            .await?
+            .can(crate::MemberPermission::CanLinkSpaces)
+        {
+            bail!("You don't have permissions to add parent-rooms");
+        }
+        if !self.is_joined() {
+            bail!("You can't update a room you aren't part of");
+        }
+        let room = self.room.clone();
+        let client = self.core.client().clone();
+
+        RUNTIME
+            .spawn(async move {
+                let Some(Ok(homeserver)) = client.homeserver().host_str().map(|h| h.try_into()) else {
+                    return Err(Error::HomeserverMissesHostname)?;
+                };
+                let mut content = SpaceParentEventContent::new(vec![homeserver]);
+                if canonical {
+                    content.canonical = true;
+                }
+                let response = room
+                    .send_state_event_for_key(
+                        &room_id,
+                        content,
+                    )
+                    .await?;
+                Ok(response.event_id.to_string())
+            })
+            .await?
+    }
+
+    pub async fn remove_parent_room(
+        &self,
+        room_id: String,
+        reason: Option<String>,
+    ) -> Result<bool> {
+        let room_id = OwnedRoomId::try_from(room_id)?;
+        if !self
+            .get_my_membership()
+            .await?
+            .can(crate::MemberPermission::CanLinkSpaces)
+        {
+            bail!("You don't have permissions to remove child-rooms");
+        }
+        if !self.is_joined() {
+            bail!("You can't update a space you aren't part of");
+        }
+        let room = self.room.clone();
+
+        RUNTIME
+            .spawn(async move {
+                let response = room
+                    .get_state_event_static_for_key::<SpaceParentEventContent, OwnedRoomId>(
+                        &room_id,
+                    )
+                    .await?;
+                let Some(raw_state) = response else {
+                    warn!("This room {} was not found in parents", room_id);
+                    return Ok(false);
```

Comment:
Yeah, either we return `true` here, too, to indicate that the room _isn't_ a parent anymore or we want this to a `bail!()` condition. But just returning false doesn't allow the caller to understand what happened or why.

I vote for also just returning `true`.

Reply:
Fixed by https://github.com/acterglobal/a3/pull/1089/commits/44593dd54a303be626a1d2380b339d03300e2ab6